### ISSUE TITLE
Would like to re-use the existing image save function to save data on cordova/phonegap

### ISSUE THREAD

**Issue by shusain** [processing/p5.js#2371]
#### Nature of issue?

- [ ] Found a bug
- [x] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core
- [ ] Data
- [ ] Events
- [x] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [x] Mobile/Tablet (touch devices)
- [ ] Desktop/Laptop
- [ ] Others (specify if possible)

- p5.js version: v0.5.16
- Web browser and version: Cordova/PhoneGap Mobile Chrome
- Operating System: Android/iOS

#### Feature enhancement details:
I'd like to add a third optional argument for the `Image#save` function to avoid triggering the download and just return the image data instead so I can use the cordova file plugins to save the data to the file system instead of the baked in browser trying to prompt the user.

https://github.com/processing/p5.js/blob/master/src/image/p5.Image.js#L763

Basically could add a `onlyReturnData` parameter (open to better names) that is false by default and if passed as true will just skip the download statement and add a return at the end of the function to return the data.

**Comment by Spongman** [comment]
you could always just call `image.canvas.toDataUrl('image/png')` to get the data.

**Comment by shusain** [comment]
Yeah fair enough I suppose the fiddling with metadata that this function does probably isn't necessary except when using the download call anyhow.  Also realized after opening this I actually want to draw the data from the canvas of a renderer I created with createGraphics instead of from an Image object sooo makes this moot for my use case anyhow... thanks for taking a look and the heads up.
